### PR TITLE
Support kind besides Minikube

### PR DESCRIPTION
**Pull request by palkra**
## Reasons for this PR:
 - We run our deployment on a KinD cluster, to which the production network is not yet compatible.
 - Many tasks of the network deployment should be moved in the cluster for a better architecture.

## Changes in this PR:
 - CA containers enroll their own admins and register initial users on the CA server container after deployment
 - Add Kubernetes Jobs to enroll users instead of doing this in the deployment script
 - Replace templating by the static internal addresses of the networks components to reduce complexity
 - Remove minikube specific commands from the mains scripts and add an extra setup script for use with minikube
 - Add a pipeline to test the network on both Minikube and KinD
 - Add fix for KinD permission errors due to read-only and protected directories
 - Add instructions to the readme for running the network on both Minikube and KinD

### CODE REVIEW

**Comment by palkra** [comment]
Change mounting folder to mount that is provided by kind. Add a script that provides the same mounts to Minikube to guarantee compatibility to both Kubernetes environments.

**Comment by palkra** [comment]
The next step is to remove the IP addresses for enrollment and registration requests to the CA servers. We should use the service names instead.

**Comment by palkra** [comment]
Changed the CAs to register initial users and enroll admin users in the containers.

File: configtx.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ Organizations:
         ID: org0MSP
 
         # MSPDir is the filesystem path which contains the MSP configuration.
-        MSPDir: tmp/hyperledger/org0/msp
+        MSPDir: /data/uc4/development/hyperledger/org0/msp
```
Comment (NikoBergemann) [inline configtx.yaml:36]: (future clean up - maybe(?)) can we replace the static directory existing on the specific mount by a configurable EnvVar?

Comment (NikoBergemann) [reply configtx.yaml:36]: TMP_FOLDER seems to exist already

**Review by NikoBergemann — APPROVED** [review]

File: configtx.yaml (hunk 1 above)
Comment (palkra) [reply configtx.yaml:36]: Good point, we will create an issue for that :+1:

Comment (palkra) [reply configtx.yaml:36]: https://github.com/upb-uc4/hlf-network/issues/49

**Review by anemonekk — APPROVED** [review]
Tested locally, works on my machine with script restartKindCluster

**Pull request merged**